From 5da42f1f90244e836020e1149592e353737c5719 Mon Sep 17 00:00:00 2001
From: Jakub Nowacki <jakubx.nowacki@intel.com>
Date: Fri, 02 Jun 2023 11:39:48 +0200
Subject: [PATCH] Shorten threshold name in RedFish log message

The threshold names in Redfish event log entries are logged with full
name, for example:
"xyz.openbmc_project.Telemetry.Trigger.Type.LowerWarning". This commit
makes change to log only the last part of the name, so in this example,
it will log only "LowerWarning" part.

Tested:
On platform, the proper threshold name is now logged.

Change-Id: I8c2907a0c627870da29612b5a3ea6a1b412109e2
Signed-off-by: Jakub Nowacki <jakubx.nowacki@intel.com>
Upstream-Status: Pending
---

diff --git a/src/trigger_actions.cpp b/src/trigger_actions.cpp
index ef4a50b..84d048f 100644
--- a/src/trigger_actions.cpp
+++ b/src/trigger_actions.cpp
@@ -95,10 +95,11 @@
         "Logging numeric trigger action to Redfish Event Log.",
         phosphor::logging::entry("REDFISH_MESSAGE_ID=%s",
                                  getRedfishMessageId()),
-        phosphor::logging::entry("REDFISH_MESSAGE_ARGS=%s,%s,%s,%f,%s,%s",
-                                 thresholdName.c_str(), triggerId.c_str(),
-                                 sensorName.c_str(), value, direction,
-                                 timestampStr.c_str()));
+        phosphor::logging::entry(
+            "REDFISH_MESSAGE_ARGS=%s,%s,%s,%f,%s,%s",
+            std::string(utils::toShortEnum(thresholdName)).c_str(),
+            triggerId.c_str(), sensorName.c_str(), value, direction,
+            timestampStr.c_str()));
 }
 
 void fillActions(
